lib/src/stream_extensions.dart

```diff
@@ -33,4 +33,24 @@ extension StreamExtensions<T> on Stream<T> {
       sink.close();
     }));
   }
+
+  /// Returns the first data or error event emitted by [this] if it emits any,
```

Comment:
Should be documented with a noun phrase (it's a getter).

So either (underplaying the asynchrony)

/// The first data or event emitted by this stream, or `null` if the stream is empty.

or mention the future,

/// A future which completes with the first event of this stream, or with `null`.
///
/// If this stream emits any event, whether a data event or an error event,
/// the future completes with the same data value or error.
/// If the stream ends without emitting any events, the future is completed with
/// the `null` value.

I think I prefer the latter.